plugins/generator-1.19.4/forge-1.19.4/procedures/compare_damagesources.java.ftl

```diff
@@ -0,0 +1 @@
+(${input$a} == ${input$b})
```

Comment:
You can't compare DamageSource like this in 1.19.4 as anyone can create DamageSource and == compares instances (pointers to objects in the memory).

You should either compare types or use DamageSource#is (preferred). 

This also partially applies to 1.19.2

Reply:
> You should either compare types or use DamageSource#is (preferred).

`DamageSource#is` doesn't exist

Reply:
![image](https://user-images.githubusercontent.com/16374228/235979215-a5f6f352-1147-4235-9009-bffb82de6744.png)

Reply:
![image](https://user-images.githubusercontent.com/16374228/235979355-d2ca4d9e-4874-4066-b6c9-78152249a650.png)

My Minecraft has it

Reply:
Mine doesn't for some reason...

<img width="2229" alt="Capture d’écran 2023-05-03 à 18 26 36" src="https://user-images.githubusercontent.com/49318596/235979615-f7fc38f8-dacd-4704-a3a5-110c74aaca81.png">

Reply:
I am checking 1.19.4. Any way, you will likely need to compare by type of damage source, not by the damage source itself, similar how you can't compare directly two itemstacks, but need to do it by item

Reply:
I don't understand, I searched Minecraft's code, and everywhere I look at, they directly compare DamageSources like this : `DamageSource.X == DamageSource.Y`, or `DamageSource.X.equals(DamageSource.Y)`

Reply:
1.19.2 checks on damagesource directly or more commonly using methods such as isMagic, isFall, ..., but 1.19.4 checks on the type. How come is this?

![image](https://user-images.githubusercontent.com/16374228/236614507-61ca6ca0-6908-4bc9-9972-e0c8ac2fc26c.png)

![image](https://user-images.githubusercontent.com/16374228/236614518-f86dd72d-3b79-42ed-aaf3-56aa5663ab1e.png)

If this does not show == will not work, then I don't know what will

Reply:
For equals case:

![image](https://user-images.githubusercontent.com/16374228/236614577-982e30fe-dc2a-4f94-8d77-645af4e423d2.png)

![image](https://user-images.githubusercontent.com/16374228/236614668-43976d53-6371-4e59-a3bd-14108e971507.png)

-----

I have lost too much time on this PR so I am leaving it now and will come back when things will work as they should as I can't assist with every single step unfortunately, due to lack of time and 40 other PRs that need my attention

Reply:
I fixed the issue with 1.19.4, but in 1.19.2, there is no DamageType stuff, and therefore, it is still required to use basic `==`